src/consensus.rs

```diff
@@ -562,13 +559,11 @@ impl <L, M> Consensus<L, M> where L: Log, M: StateMachine {
             let message =
                 messages::command_response_not_leader(&self.peers[&self.follower_state.leader.unwrap()]);
             actions.client_messages.push((from, message));
-        } else {
+        } else if let Ok(entry) = request.get_entry() {
             let prev_log_index = self.latest_log_index();
             let prev_log_term = self.latest_log_term();
             let term = self.current_term();
             let log_index = prev_log_index + 1;
-            // TODO: This is probably not exactly safe.
-            let entry = request.get_entry().unwrap();
```

Comment:
Again, I'm not sure crashing is such a bad thing right now.  Failures to write or read log entries are a *really* bad failure, because they could lead to data loss / consistency errors.  I don't think crashing is such a bad option, at least right now.  Recovering from these kind of failures will potentially require catching the replica up from a peer, or in the worst case (out of disk space, for instance), taking the replica offline.